src/frontend/contexts/EarlyAccessContext.tsx

```diff
@@ -0,0 +1,75 @@
+import {createContext, useContext} from 'react';
+import {createStore, useStore, type StoreApi} from 'zustand';
+import {
+  createJSONStorage,
+  persist as createPersistedState,
+} from 'zustand/middleware';
+import {MMKVZustandStorage} from '../hooks/persistedState/createPersistedState';
+
+// NOTE: Do not change!
+export const STORAGE_KEY = 'EarlyAccessStore' as const;
+
+type EarlyAccessState = {
+  isEarlyAccessEnabled: boolean;
+};
+
+type EarlyAccessActions = {
+  setEarlyAccessEnabled: (enabled: boolean) => void;
+};
+
+function createInitialState(): EarlyAccessState {
+  return {isEarlyAccessEnabled: false};
+}
+
+export function createEarlyAccessStore({persist} = {persist: false}) {
+  let store: StoreApi<EarlyAccessState>;
+
+  if (persist) {
+    store = createStore(
+      createPersistedState(createInitialState, {
+        name: STORAGE_KEY,
+        storage: createJSONStorage(() => MMKVZustandStorage),
+        version: 0,
+      }),
+    );
+  } else {
+    store = createStore(createInitialState);
+  }
+
+  const actions: EarlyAccessActions = {
+    setEarlyAccessEnabled: (enabled: boolean) => {
+      store.setState({isEarlyAccessEnabled: enabled});
+    },
+  };
+
+  return {instance: store, actions};
+}
+
+export type EarlyAccessStore = ReturnType<typeof createEarlyAccessStore>;
+
+const EarlyAccessStoreContext = createContext<EarlyAccessStore | null>(null);
+
+export const EarlyAccessStoreProvider = EarlyAccessStoreContext.Provider;
+
+function useEarlyAccessStoreContext() {
+  const value = useContext(EarlyAccessStoreContext);
+  if (!value) throw new Error('EarlyAccessStoreProvider missing');
+  return value;
+}
+
+export function useEarlyAccessState<T>(
+  selector: (s: EarlyAccessState) => T,
+): T {
+  const {instance} = useEarlyAccessStoreContext();
+  return useStore(instance, selector);
+}
+
+export function useIsEarlyAccessEnabled(): boolean {
+  const {instance} = useEarlyAccessStoreContext();
+  return useStore(instance).isEarlyAccessEnabled;
+}
```

Comment:
I don't think we should use this (or at least use it this way). This hook uses the entire store, and then returns a value from that store. While there is only one value in the store now, if other values were added, this hook would cause re-renders eveytime there is a change in state even if that state is unrelated to what this hook is returning